mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -1447,7 +1447,7 @@ private class LogisticAggregator(
       label: Double): Unit = {
 
     val localFeaturesStd = bcFeaturesStd.value
-    val localCoefficients = bcCoefficients.value
+    val localCoefficients = bcCoefficients.value.toArray
```

Comment:
The above check got us into trouble because if we don't add `@transient lazy val` then we'll serialize the coefficients. The call to `toArray` is really just a small bit of pointer indirection, and while I agree it is not _great_ to call it every time, the extra function call should pale in comparison to the `O(numClasses * numFeatures)` ops we do in the method. 

That said, I'm ok with either solution, just wanted to point out the pros/cons of each. Let me know what you think, thanks for reviewing!